File: Frontend/CompilerFrontEnd/Operations/Detecting/DetectRequest.cs
namespace CompilerFrontEnd.Operations.Detecting;

public class DetectRequest
{
    public String BuildPath { get; set; }
    
    public String Repository { get; set; }
    
    public String RepositoryName { get; set; }

    public DetectRequest(string buildPath, string repository, string repositoryName)
    {
        BuildPath = buildPath;
        Repository = repository;
        RepositoryName = repositoryName;
    }
}

File: detecter/src/main/java/Model/DetectRequest.java
package Model;

public class DetectRequest {

  public static final String TYPE = "DETECT";

  private String BuildPath;


  private String Repository;

  private String RepositoryName;

  public DetectRequest(String buildPath, String repository,
                       String respositoryName) {
    this.BuildPath = buildPath;
    this.Repository = repository;
    this.RepositoryName = respositoryName;
  }

  public String getBuildPath() {
    return BuildPath;
  }

  public void setBuildPath(String buildPath) {
    this.BuildPath = buildPath;
  }

  public String getRepository() {
    return Repository;
  }

  public void setRepository(String repository) {
    this.Repository = repository;
  }

  public String getRepositoryName() {
    return RepositoryName;
  }

  public void setRepositoryName(String repositoryName) {
    this.RepositoryName = repositoryName;
  }

}
